pyanaconda/packaging/__init__.py

```diff
@@ -290,7 +290,11 @@ def selectGroup(self, groupid, default=True, optional=False):
         if grp in self.data.packages.excludedGroupList:
             self.data.packages.excludedGroupList.remove(grp)
 
-        self.data.packages.groupList.append(grp)
+        # If the group is part of the environment, no need to duplicate it in
+        # the group list
+        if self.data.packages.environment and \
+                groupid not in self.environmentGroups(self.data.packages.environment):
+            self.data.packages.groupList.append(grp)
```

Comment:
This is ok because with this patch the environment gets added to the ksdata instead of just its groups, and the selected environment gets added to the payload via dnf.Base.environment_install or yum.selectEnvironment. I'll change the comment to "If the group is part of the environment, it is already selected as part of the environment. Do not duplicate it in the group list"